test_scripts/Polices/build_options/039_ATF_P_TC_PTU_HMI_Level_Affected_Apps_FULL_LIMITED_HTTP.lua

```diff
@@ -111,97 +116,105 @@ function Test:TestStep_RegisterFirstApp()
-          {
-            requestType = "HTTP",
-            fileName = "ptu2app.json",
-          },"files/ptu2app.json")
-        self.mobileSession1:ExpectResponse(CorIdSystemRequest, {success = true, resultCode = "SUCCESS"})
-      end)
-    EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate")
-    :ValidIf(function(exp,data)
-        if
-        (exp.occurences == 1 or exp.occurences == 2) and
-        data.params.status == "UP_TO_DATE" then
-          return true
-        end
-        if
-        exp.occurences == 1 and
-        data.params.status == "UPDATING" then
-          return true
-        end
-        return false
-        end):Times(Between(1,2))
-      self.mobileSession1:ExpectNotification("OnPermissionsChange")
-      -- Expect after updating HMI status will not change
-      self.mobileSession1:ExpectNotification("OnHMIStatus"):Times(0)
-    end
-
-    --[[ Postconditions ]]
-    commonFunctions:newTestCasesGroup("Postconditions")
-
-    function Test.Postcondition_RemovePTUfiles()
-      os.remove(ptu_first_app_registered)
-      os.remove(ptu_second_app_registered)
-    end
-    function Test.Postcondition_Stop_SDL()
-      StopSDL()
-    end
-
-    return Test
+    end)
+
+  EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UPDATE_NEEDED"}, {status = "UPDATING"}):Times(2)
+end
+
+function Test:TestStep_ActivateSecondAppInLimited()
+  commonSteps:ActivateAppInSpecificLevel(self,HMIAppID,"LIMITED") -- function is working only for activate application in FULL
+  self.mobileSession:ExpectNotification("OnHMIStatus", {hmiLevel = "LIMITED", audioStreamingState = "AUDIBLE", systemContext = "MAIN"})
+  :Do(function(_,data) print("SDL -> MOB1: OnHMIStatus, level: " .. data.payload.hmiLevel) end)
+
+  self.mobileSession1:ExpectNotification("OnHMIStatus", {hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+  :Do(function(_,data) print("SDL -> MOB2: OnHMIStatus, level: " .. data.payload.hmiLevel) end)
+end
+
+function Test:TestStep_UpdatePolicyAfterAddSecondApp_ExpectOnHMIStatusNotCall()
+
+  local CorIdSystemRequest = self.mobileSession1:SendRPC("SystemRequest",
+    {
+      requestType = "HTTP",
+      fileName = "ptu2app.json",
+    },"files/ptu2app.json")
+  self.mobileSession1:ExpectResponse(CorIdSystemRequest, {success = true, resultCode = "SUCCESS"})
+
+  EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate", {status = "UP_TO_DATE"})
+
+  self.mobileSession1:ExpectNotification("OnPermissionsChange")
+  -- Expect after updating HMI status will not change
+  self.mobileSession1:ExpectNotification("OnHMIStatus"):Times(0)
+  self.mobileSession:ExpectNotification("OnHMIStatus"):Times(0)
+  commonTestCases:DelayedExp(10000)
+end
+
+--[[ Postconditions ]]
+commonFunctions:newTestCasesGroup("Postconditions")
+
+function Test.Postcondition_RemovePTUfiles()
+  os.remove(ptu_first_app_registered)
+  os.remove(ptu_second_app_registered)
+end
+
+function Test.Postcondition_Stop_SDL()
+  StopSDL()
+end
+
+return Test
```

Comment:
@istoimenova, Since the only one issue with SDL.OnStatusUpdate(UPDATE_NEEDED) still remains, is it possible to rework the script with connecttest_resumption approach in order to eliminate this issue?